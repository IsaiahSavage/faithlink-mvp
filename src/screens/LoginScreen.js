import React, { useState } from 'react';
import {
  View,
  Text,
  StyleSheet,
  KeyboardAvoidingView,
  Platform,
  ActivityIndicator,
} from 'react-native';
import { TextInput, Button } from 'react-native-paper';
import { FIREBASE_AUTH } from '../../firebase/firebaseConfig';
import { signInWithEmailAndPassword } from 'firebase/auth';

const LoginScreen = ({ navigation }) => {
  /** STATE VARIABLES */
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [hidePass, setHidePass] = useState(true);
  const [loading, setLoading] = useState(false);

  const auth = FIREBASE_AUTH;

  const signIn = async () => {
    setLoading(true);
    try {
      await signInWithEmailAndPassword(auth, email, password);
    } catch (error) {
      console.log(error);
      alert('Login failed:' + error.message || 'Unknown error');
    } finally {
      setLoading(false);
    }
  };

  return (
    <KeyboardAvoidingView
      style={styles.container}
      behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
    >
      <Text style={styles.title}>Login</Text>
      <View style={styles.inputContainer}>
        <TextInput
          style={styles.input}
          onChangeText={(text) => setEmail(text)}
          value={email}
          label="Email"
          autoComplete="email"
          keyboardType="email-address"
          autoCapitalize="none"
          underlineColor="transparent"
          theme={{ colors: { primary: '#002857' } }}
          returnKeyType="next"
          onSubmitEditing={() => {
            this.PasswordInput.focus();
          }}
          blurOnSubmit={false}
        />
        <TextInput
          style={styles.input}
          onChangeText={(text) => setPassword(text)}
          label="Password"
          value={password}
          autoCapitalize="none"
          textContentType="password"
          secureTextEntry={hidePass ? true : false}
          autoCorrect={false}
          ref={(input) => {
            this.PasswordInput = input;
          }}
          underlineColor="transparent"
          theme={{ colors: { primary: '#002857' } }}
          right={
            <TextInput.Icon icon="eye" onPress={() => setHidePass(!hidePass)} />
          }
        />
        {loading ? (
          <ActivityIndicator size={'large'} color={'#002857'} />
        ) : (
          <>
            <Button
              mode="contained"
              style={styles.btnContainer}
              buttonColor="#002857"
              labelStyle={styles.loginText}
              dark={true}
              onPress={signIn}
            >
              Login
            </Button>
            <Button
              mode="text"
              style={styles.btnContainer}
              labelStyle={styles.loginText}
              onPress={() => navigation.navigate('Register')}
            >
              Create an Account
            </Button>
          </>
        )}
      </View>
    </KeyboardAvoidingView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
  },
  title: {
    color: '#002857',
    fontSize: 30,
    marginTop: 50,
  },
  inputContainer: {
    justifyContent: 'center',
    flex: 2,
  },
  input: {
    fontSize: 20,
    minWidth: 300,
    marginBottom: 20,
    backgroundColor: '#E8E8E8',
  },
  btnContainer: {
    alignSelf: 'center',
    width: '80%',
    alignItems: 'center',
    borderRadius: 100,
    marginTop: 20,
  },
  loginText: {
    fontSize: 20,
    width: '100%',
    height: 'auto',
  },
});

export default LoginScreen;
