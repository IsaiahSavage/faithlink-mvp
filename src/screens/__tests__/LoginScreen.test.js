import React from 'react';
import { render } from '@testing-library/react-native';

import LoginScreen from '../LoginScreen';
describe('Login Screen',() =>{

    //does the page render

    describe('LoginScreen', () => {
        it('renders correctly', () => {
          render(<LoginScreen />);
        });

    //does it match the snapshot
    it('matches the snapshot', () => {
        const { toJSON } = render(<LoginScreen />);
        expect(toJSON()).toMatchSnapshot();
    });

    //is the header the same?
    it('header is the same', () => {
        const { getByText } = render(<LoginScreen />);
        const title = getByText(/Login/);
        expect(title).toBeTruthy();
      });

    //Does the create account button take you to the registration page?

    //Does the test login actually log you in?
})});