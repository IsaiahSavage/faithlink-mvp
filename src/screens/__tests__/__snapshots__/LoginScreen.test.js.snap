// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen matches the snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <Text
    style={
      {
        "color": "#002857",
        "fontSize": 30,
        "marginTop": 50,
      }
    }
  >
    Login
  </Text>
  <View
    style={
      {
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "rgba(231, 224, 236, 1)",
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          {
            "backgroundColor": "#E8E8E8",
            "marginBottom": 20,
            "minWidth": 300,
          },
        ]
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "bottom": 0,
            "height": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "transform": [
              {
                "scaleY": 0.5,
              },
            ],
            "zIndex": 1,
          }
        }
        testID="text-input-underline"
      />
      <View
        onLayout={[Function]}
        style={
          [
            {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            {
              "minHeight": 56,
            },
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": [
                {
                  "translateX": 0,
                },
              ],
              "width": 750,
              "zIndex": 3,
            }
          }
        >
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            onLayout={[Function]}
            onTextLayout={[Function]}
            style={
              {
                "color": "#002857",
                "fontFamily": "System",
                "fontSize": 20,
                "fontWeight": undefined,
                "left": 0,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "maxWidth": 73,
                "opacity": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "absolute",
                "textAlign": "left",
                "top": 30,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
            testID="text-input-flat-label-active"
          >
            Email
          </Text>
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            style={
              {
                "color": "rgba(73, 69, 79, 1)",
                "fontFamily": "System",
                "fontSize": 20,
                "fontWeight": undefined,
                "left": 0,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "maxWidth": 73,
                "opacity": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "absolute",
                "textAlign": "left",
                "top": 30,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
            testID="text-input-flat-label-inactive"
          >
            Email
          </Text>
        </View>
        <TextInput
          autoCapitalize="none"
          autoComplete="email"
          blurOnSubmit={false}
          cursorColor="#002857"
          editable={true}
          keyboardType="email-address"
          maxFontSizeMultiplier={1.5}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onSubmitEditing={[Function]}
          placeholder=" "
          placeholderTextColor="rgba(73, 69, 79, 1)"
          returnKeyType="next"
          selectionColor="#002857"
          style={
            [
              {
                "margin": 0,
              },
              {
                "height": 56,
              },
              {
                "paddingBottom": 4,
                "paddingTop": 24,
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontSize": 20,
                "fontWeight": undefined,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "minWidth": 65,
                "paddingLeft": 16,
                "paddingRight": 16,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              [
                {},
              ],
              undefined,
            ]
          }
          testID="text-input-flat"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "rgba(231, 224, 236, 1)",
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          {
            "backgroundColor": "#E8E8E8",
            "marginBottom": 20,
            "minWidth": 300,
          },
        ]
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "bottom": 0,
            "height": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "transform": [
              {
                "scaleY": 0.5,
              },
            ],
            "zIndex": 1,
          }
        }
        testID="text-input-underline"
      />
      <View
        onLayout={[Function]}
        style={
          [
            {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            {
              "minHeight": 56,
            },
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": [
                {
                  "translateX": 0,
                },
              ],
              "width": 750,
              "zIndex": 3,
            }
          }
        >
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            onLayout={[Function]}
            onTextLayout={[Function]}
            style={
              {
                "color": "#002857",
                "fontFamily": "System",
                "fontSize": 20,
                "fontWeight": undefined,
                "left": 0,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "maxWidth": 73,
                "opacity": 0,
                "paddingLeft": 16,
                "paddingRight": 56,
                "position": "absolute",
                "textAlign": "left",
                "top": 30,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
            testID="text-input-flat-label-active"
          >
            Password
          </Text>
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            style={
              {
                "color": "rgba(73, 69, 79, 1)",
                "fontFamily": "System",
                "fontSize": 20,
                "fontWeight": undefined,
                "left": 0,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "maxWidth": 73,
                "opacity": 0,
                "paddingLeft": 16,
                "paddingRight": 56,
                "position": "absolute",
                "textAlign": "left",
                "top": 30,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
            testID="text-input-flat-label-inactive"
          >
            Password
          </Text>
        </View>
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          cursorColor="#002857"
          editable={true}
          maxFontSizeMultiplier={1.5}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder=" "
          placeholderTextColor="rgba(73, 69, 79, 1)"
          secureTextEntry={true}
          selectionColor="#002857"
          style={
            [
              {
                "margin": 0,
              },
              {
                "height": 56,
              },
              {
                "paddingBottom": 4,
                "paddingTop": 24,
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontSize": 20,
                "fontWeight": undefined,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "minWidth": 65,
                "paddingLeft": 16,
                "paddingRight": 56,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              {
                "marginRight": 40,
                "paddingRight": 16,
              },
              undefined,
            ]
          }
          testID="text-input-flat"
          textContentType="password"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          [
            {
              "alignItems": "center",
              "height": 24,
              "justifyContent": "center",
              "position": "absolute",
              "width": 24,
            },
            {
              "right": 16,
              "top": 16,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "transparent",
              "borderRadius": 20,
              "height": 40,
              "margin": 0,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
              "width": 40,
            }
          }
          testID="right-icon-adornment-container-outer-layer"
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "transparent",
                "borderColor": "rgba(121, 116, 126, 1)",
                "borderRadius": 20,
                "borderWidth": 0,
                "elevation": 0,
                "flex": 1,
                "overflow": "hidden",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="right-icon-adornment-container"
          >
            <View
              accessibilityComponentType="button"
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityTraits="button"
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              centered={true}
              collapsable={false}
              focusable={true}
              hitSlop={
                {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  [
                    {
                      "alignItems": "center",
                      "flexGrow": 1,
                      "justifyContent": "center",
                    },
                    {
                      "borderRadius": 20,
                    },
                  ],
                ]
              }
              testID="right-icon-adornment"
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  [
                    {
                      "color": "rgba(73, 69, 79, 1)",
                      "fontSize": 24,
                    },
                    [
                      {
                        "lineHeight": 24,
                        "transform": [
                          {
                            "scaleX": 1,
                          },
                        ],
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                    ],
                    {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                󰈈
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "#002857",
          "borderRadius": 100,
          "marginTop": 20,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "width": "80%",
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#002857",
            "borderColor": "transparent",
            "borderRadius": 100,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": undefined,
            "minWidth": 64,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 100,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      {
                        "color": "#ffffff",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      {
                        "fontSize": 20,
                        "height": "auto",
                        "width": "100%",
                      },
                    ],
                  ],
                ]
              }
              testID="button-text"
            >
              Sign In
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderRadius": 100,
          "marginTop": 20,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "width": "80%",
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 100,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": undefined,
            "minWidth": 64,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 100,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 12,
                      },
                      undefined,
                      false,
                      {
                        "color": "rgba(103, 80, 164, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      {
                        "fontSize": 20,
                        "height": "auto",
                        "width": "100%",
                      },
                    ],
                  ],
                ]
              }
              testID="button-text"
            >
              Create an Account
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
